lib/reporter.js

```diff
@@ -15,20 +15,33 @@ class JunitReporter extends events.EventEmitter {
         super()
 
         this.baseReporter = baseReporter
+        this.suitTitles = {}
         this.config = config
         this.options = options
         this.suiteNameRegEx = this.options.suiteNameFormat instanceof RegExp
             ? this.options.suiteNameFormat
             : /[^a-z0-9]+/
 
+        this.on('runner:start', function (runner) {
+            this.suitTitles[runner.cid] = {}
+        })
+
+        this.on('suite:start', function (suite) {
+            this.suitTitles[suite.cid][suite.uid] = suite.fullTitle
```

Comment:
I believe this is only available in Mocha, how do we deal with this if someone uses a different framework?

Reply:
Yes, I hadn't thought about it. At least 'parent' attribute is available in Jasmine and Mocha so I can join fullTitle directly in this function. Let me extend it.
Also seems that parent suite always starts before. So I can suggest by the time child suite will start I will already have a name for parent.